src/virtualenv/run/plugin/base.py

```diff
@@ -11,7 +15,11 @@ class PluginLoader(object):
 
     @classmethod
     def entry_points_for(cls, key):
-        return OrderedDict((e.name, e.load()) for e in cls.entry_points().select(group=key))
+        try:
```

Comment:
Instead of try/catch can we make it explicit based on the version? I want it so that an AttributeError on 3.10 is now silently swallowed.

Reply:
We would need to get the version of importlib_metadata.

Reply:
We should only need importlib_metadata where we don't have importlib.metadata

Reply:
Yes that is correct. But we would still need to check the version then.

Reply:
The if would need to be (pseudo-code):

```
if PYTHON_VERSION >= 3.10 or (PYTHON_VERSION < 3.8 and IMPORTLIB_METADATA_VERSION < 3.6):
```

I don't like that very much.

Reply:
I prefer it personally

Reply:
I wouldn't know how to get the importlib metadata version.

This works:

```
>>> importlib_metadata.version('importlib_metadata')
'4.8.2'
```

But I would need to parse it with packaging.version to get a comparable object.

Reply:
You can just split by . and cast the first two element to int, no?

Reply:
I can do that, but that seems rather fragile. What if the next version is 5b2? I am hesitant to implement our own version parsing logic just so we can do the check. I can certainly try.

Reply:
I am not happy about the code, but I've done it.